odoo/addons/test_lint/tests/_odoo_checker_gettext.py

```diff
@@ -49,10 +50,19 @@ class OdooBaseChecker(BaseChecker):
             'gettext-repr',
             'Don\'t use %r to automatically insert quotes in translation strings. Quotes can be different depending on the language: they must be part of the translated string.',
         ),
+        'E8507': ('Static string passed to %s without gettext call.',
+                  'missing-gettext',
+                  'Ensure that all static strings passed to certain constructs are wrapped in a gettext call.')
     }
 
-    @only_required_for_messages('gettext-variable', 'gettext-placeholders', 'gettext-repr')
+    functions = ['UserError', 'ValidationError', 'AccessError', 'AccessDenied', 'MissingError']
+
+    @only_required_for_messages('missing-gettext', 'gettext-variable', 'gettext-placeholders', 'gettext-repr')
     def visit_call(self, node):
+        file_path = self.linter.current_file
+        if os.sep + "test_" in file_path:
```

Comment:
It would be better to check if the file is in the `tests` package rather than checking if it starts with `test_`.

Reply:
we have files that are in `addons/test_something/something.py` so matching the prefix `/test` would match both: `addons/test_*/*.py` and `/tests/*.py`

wdyt?

Reply:
I think that the prefix is too broad. You could have business code like `some_module/models/test_results.py` (example: a laboratory or a physical tests in a gym). You can always check some of the test modules (their controllers, models) as they should be written like business code.

Reply:
I see
But modules starting with `test_{module_name}` aren't exported to be translated so translating them would make no sense. Also, if there are some cases in test modules that we want to translate, enabling lint checks would require us to make all of them translated.

Reply:
Then you need to find the module name from the path instead of just testing with a simple `in`. There is a function `get_module_root` to get the path to the module from a file. What do you think of the following?

```python
if (
  "test" in file_path  # pre-filter
  and (module_path := get_module_root(file_path)
  # test module or tests package in a module
  and (base_name(module_path).startswith("test_") or file_path[len(module_path) + 1:].startswith("tests" + os.sep))
)
```